### PR TITLE
solution

### PR DESCRIPTION
**Pull request by vitvadym**

### CODE REVIEW

File: src/herbivoresAndCarnivores.js [hunk 1]
```diff
@@ -1,15 +1,47 @@
...
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, damage = 50) {
+    super(name);
+    this.damage = damage;
+  }
```
Comment (33Orange) [inline src/herbivoresAndCarnivores.js:34]: I do not see in the conditions of the task that the damage of a predator can be different. Let's do everything according to the conditions
```suggestion

```

File: src/herbivoresAndCarnivores.js [hunk 2]
```diff
@@ -1,15 +1,47 @@
...
+
+  bite(animal) {
+    animal instanceof Herbivore && !animal.hidden
+      ? animal.health -= this.damage
+      : animal.health = this.health;
```
Comment (33Orange) [inline src/herbivoresAndCarnivores.js:39]: Why you use ternary operator here?
We use ternary operator only if we do smth in every case, but here you do smth only if condition `true`.

```suggestion
    if (animal instanceof Herbivore && !animal.hidden) {
      animal.health -= this.damage
    }
```

**Review by 33Orange — CHANGES_REQUESTED** [review]
Good job.

A few moments to fix

File: src/herbivoresAndCarnivores.js [hunk 3]
```diff
@@ -1,15 +1,45 @@
...
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static removeDiedAmimals() {
```
Comment (silchyn) [inline src/herbivoresAndCarnivores.js:10]:
```suggestion
  static removeDeadAmimals() {
```

**Review by silchyn — APPROVED** [review]

**Pull request closed without merging**